listener/listener.go

```diff
@@ -260,6 +272,10 @@ func (l *Listener) Listen(ctx context.Context, handler message.Handler, topicNam
 	// Create a handle class that has that function
 	listenerHandle := subReceiver.Listen(ctx, servicebus.HandlerFunc(
 		func(ctx context.Context, msg *servicebus.Message) error {
+			if l.lockRenewalInterval != nil {
+				renewer := peeklock.RenewPeriodically(ctx, *l.lockRenewalInterval, sub, msg)
+				defer renewer.Stop()
```

Comment:
why the stop idiom rather than doing  
```
renewerecontext, cancel = context.WithCancelation(ctx)
defer cancel()
peeklock.RenewPeriodically(renewercontext)
```
My understanding is that the cancel only cancels renewercontext and not ctx.

Reply:
ah you are doing that inside renwew lock but why return a wholre renewer rather than just a cancel function. Maybe you're thinkiing of renewer having something else eventually?

Reply:
I started with the struct, and I got there this way.
Plan for the renewer would be to be able to batch concurrent message renewal to avoid increasing the number of calls too fast when increasing concurrency. I'm just not sure how it will look like. since this context is created inside the renewer, I thought it was cleaner to keep it private.